localstack/services/sqs/provider.py

```diff
@@ -267,10 +268,24 @@ def arn(self) -> str:
         return f"arn:aws:sqs:{self.key.region}:{self.key.account_id}:{self.key.name}"
 
     def url(self, context: RequestContext) -> str:
-        """Return queue URL using either SQS_PORT_EXTERNAL (if configured), or based on the 'Host' request header"""
+        """Return queue URL using either SQS_PORT_EXTERNAL (if configured), the SQS_ENDPOINT_STRATEGY (if configured)
+        or based on the 'Host' request header"""
+
         host_url = context.request.host_url
-        if config.SQS_PORT_EXTERNAL:
-            host_url = external_service_url("sqs")
+
+        if config.SQS_ENDPOINT_STRATEGY == "domain":
+            # queue.localhost.localstack.cloud:4566/000000000000/my-queue (us-east-1)
+            # or us-east-2.queue.localhost.localstack.cloud:4566/000000000000/my-queue
+            region = "" if self.key.region == "us-east-1" else self.key.region + "."
```

Comment:
This is some interesting AWS behaviour^^

Reply:
true not very clean! it's just how AWS behaves.

so it's either "queue.amazonaws.com" for us-east-1 queues, and "<region>.queue.amazonaws.com" for all others. :shrug: